docs/resources/cluster.md

```diff
@@ -0,0 +1,59 @@
+# kind_cluster
+
+Provides a Kind cluster resource. This can be used to create and delete Kind
+clusters. It does NOT support modification to an existing kind cluster.
+
+## Example Usage
+
+```hcl
+# Create a kind cluster of the name "test-cluster" with default kubernetes
+# version specified in kind
+# ref: https://github.com/kubernetes-sigs/kind/blob/master/pkg/apis/config/defaults/image.go#L21
+resource "kind_cluster" "default" {
+    name = "test-cluster"
+}
+```
+
+To override the node image used:
+
+```hcl
+provider "kind" {}
+
+# Create a cluster with kind of the name "test-cluster" with kubernetes version v1.16.1
+resource "kind_cluster" "default" {
+    name = "test-cluster"
+    node_image = "kindest/node:v1.16.1"
+}
+```
+
+To override the default kind config:
+
+```hcl
+provider "kind" {}
+
+# creating a cluster with kind of the name "test-cluster" with kubernetes version v1.18.4 and two nodes
+resource "kind_cluster" "default" {
+    name = "test-cluster"
+    node_image = "kindest/node:v1.18.4"
+    kind_config =<<KIONF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+KIONF
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) The kind name that is given to the created cluster.
+* `node_image` - (Optional) The node_image that kind will use (ex: kindest/node:v1.15.3).
+* `wait_for_ready` - (Optional) Defines wether or not the provider will wait for the control plane to be ready. Defaults to false.
+* `kind_config` - (Optional) The kind_config that kind will use.
+* `kubeconfig` - (Optional) Kubeconfig set after the the cluster is created.
```

Comment:
Not sure how to list these, as all arguments below are set from the provider, when the cluster is created.

Maybe we can mark them as `Computed` instead of `Optional`

Reply:
Basically like this
```
* `kubeconfig` - (Computed) Kubeconfig set after the the cluster is created.
* `kubeconfig_path` (Computed) Kubeconfig path set after the the cluster is created.
* `client_certificate` (Computed) Client certificate for authenticating to cluster.
* `client_key` (Computed) Client key for authenticating to cluster.
* `cluster_ca_certificate` (Computed) Client verifies the server certificate with this CA cert.
* `endpoint` (Computed) Kubernetes API server endpoint.
```

Reply:
@tehcyx After reading through https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema?tab=doc#Schema

I believe the right way is not to add these arguments in the doc, as they not settable at all by users. I'll remove the above 6 arguments from this PR. Pls let me know if you want it done otherwise.

```
	// When Computed is set without Optional, the attribute is not settable
	// in configuration at all and so SchemaConfigModeAttr is the automatic
	// behavior, and SchemaConfigModeBlock is not permitted.
```

Reply:
Fair enough, fine with me.